src/main/java/jenkins/plugins/openstack/compute/internal/Openstack.java

```diff
@@ -155,28 +156,69 @@ public Openstack(@Nonnull final OSClient client) {
         return clientProvider.getExpires();
     }
 
+    /** Cache results of OSClient.networking().network().list() between threads as it doesn't change much. */
+    private final CacheableData<List<? extends Network>> cachedNetworkingNetworkListProvider = new CachedData<List<? extends Network>>(
+            15) {
+        @Override
+        protected List<? extends Network> calculate() {
+            LOGGER.log(Level.FINER, "{0}:{1} calling networking().network().list()", new Object[]{ this, Thread.currentThread() });
+            return Collections.unmodifiableList(clientProvider.get().networking().network().list());
+        }
+    };
+
     public @Nonnull Collection<? extends Network> getSortedNetworks() {
-        List<? extends Network> nets = clientProvider.get().networking().network().list();
+        List<? extends Network> nets = new ArrayList<>(cachedNetworkingNetworkListProvider.get());
```

Comment:
Why we wrapping the collection to unmodifiable one _and_ creating a copy of it? One or the other technique should do unless I miss something.

Reply:
It's important that any cached data is immutable, as otherwise the cache could be corrupted - that's why the cached data is wrapped as an unmodifiable list.

My thinking here was that the caching would operate at the lowest level, caching the raw result from Openstack4J, and then the various methods in Openstack.java would sort/filter that data to provide the answers they need.

I agree that creating a second copy is not ideal.  It's (much) faster than going back to the Openstack4J layer and re-requesting the data but I guess that it could be improved further.

Reply:
In the past, I used to trust openstack4j that data provided to us will not longer be changed by the library. So I guess we are making sure individual consumers within the plugin will not temper with it. Wrapping whatever os4j return to unmodifiable collection, caching that and passing it along to the plugin seems safe then.